py/server/tests/test_table.py

```diff
@@ -724,6 +725,48 @@ def test_python_field_access(self):
         self.assertIn("AAPL-GOOG", html_output)
         self.assertIn("2000", html_output)
 
+    def test_vectorization_exceptions(self):
+        t = empty_table(1)
+
+        @dh_vectorize
+        def vectorized_func(p1, p2):
+            return p1 + p2
+
+        def auto_func(p1, p2):
+            return p1 + p2
+
+        @dh_vectorize
+        def no_param_func():
+            return random.randint(0, 100)
+
+        with self.subTest("parameter number mismatch"):
+            with self.assertRaises(DHError) as cm:
+                t1 = t.update("X = vectorized_func(i)")
+            self.assertRegex(str(cm.exception), r".*vectorized.*count.*mismatch", )
+
+            with self.assertRaises(DHError) as cm:
+                t1 = t.update("X = auto_func(i)")
+
+        with self.subTest("can't cast return value"):
+            with self.assertRaises(DHError) as cm:
+                t1 = t.update("X = (float)vectorized_func(i, ii)")
+            self.assertIn("can't be casted", str(cm.exception))
+
+        with self.subTest("not be part of another expression"):
+            with self.assertRaises(DHError) as cm:
+                t1 = t.update("X = k + vectorized_func(i, ii)")
+            self.assertIn("in another expression", str(cm.exception))
```

Comment:
I think so, 
```
    private void checkVectorizability(MethodCallExpr n, Expression[] expressions, Class<?>[] expressionTypes,
            CallableWrapper callableWrapper) {

        callableWrapper.parseSignature();

        // Python vectorized functions(numba, DH) return arrays of primitive/Object types. This will break the generated
        // expression evaluation code that expects singular values. This check makes sure that numba/dh vectorized
        // functions must be used alone as the entire expression.
        n.getParentNode().ifPresent(parent -> {
            if (parent.getClass() == CastExpr.class) {
                throw new RuntimeException(
                        "The return values of Python vectorized function can't be casted: " + parent);
            }
            throw new RuntimeException("Python vectorized function can't be used in another expression: " + parent);
        });

        List<Class<?>> params = callableWrapper.getParamTypes();
        if (params.size() != expressions.length) {
            throw new RuntimeException("Python vectorized function argument count mismatch: " + params.size()
                    + " vs. " + expressions.length);
        }
        for (int i = 0; i < expressions.length; i++) {
            if (!(expressions[i] instanceof NameExpr)) {
                throw new RuntimeException(
                        "Python vectorized function arguments can only be columns: " + expressions[i]);
            }
            if (!isSafelyCoerceable(expressionTypes[i], params.get(i))) {
                throw new RuntimeException("Python vectorized function argument type mismatch: "
                        + expressionTypes[i].getSimpleName() + " -> " + params.get(i).getSimpleName());
            }
        }
    }
```